### DISCUSSION TITLE
Wrangling

### DISCUSSION THREAD

**Discussion by Anggitnoviee** [teamalgoritma/community#284] [Q&A] [answered]
bagaimana cara membuat kolom baru dari data jam menggunakan R, untuk mendefinisikan jam 06:30 - 13:30 adalah shift 1, 13:30 - 20:30 adalah shift 2, dan 20:30 - 06:30 shift 3. mohon bantuannya terimakasih

```
data_clean_lab <- data_clean_lab %>% 
  mutate(data_clean_lab,
      shift = case_when(waktu >= 06:30:00 ~ "shift 1",
                                    waktu >= 13:31:00 & waktu <= 20:30:00 ~ "shift 2",
                                    waktu >= 20:31:00 ~ "shift 3"))
```

saya sudah mencoba itu tp beberapa terdapat NA dan jawabannya tidak sesuai, apakah ada kesalahan?

**Comment by IrfanChairurrachman** [comment, marked as answer]
Dear @Anggitnoviee 

Hal yang ingin Anda lakukan adalah mendefinisikan shift 1, shift 2, dan shift 3 jika shift tersebut termasuk dalam rentang waktu tertentu. Hal pertama yang harus diperhatikan adalah memastikan bahwa kolom `waktu` telah dalam tipe data `Datetime`. Kemudian, perbandingan dapat bekerja untuk kelas yang sama, maka dari itu `06:30:00` seharusnya juga diubah menjadi tipe data `Datetime`. Dalam hal ini dapat digunakan fungsi `hour(x)` dimana x adalah rentang waktu tiap shiftnya.

referensi: https://stackoverflow.com/questions/66020654/mutate-case-when-date-time-data-in-r

Sekian, semoga jawaban kami membantu.

**Reply by Anggitnoviee** [reply]
Terimakasi
